src/dropdown/dropdown.component.ts

```diff
@@ -387,36 +387,36 @@ export class Dropdown implements OnInit, AfterContentInit, AfterViewInit, OnDest
 		const isUpdate = event => event && event.isUpdate;
 
 		this.view.select.subscribe(event => {
-			if (this.type === "single" && !isUpdate(event)) {
-				this.closeMenu();
-				if (event.item && event.item.selected) {
-					if (this.itemValueKey) {
-						this.propagateChange(event.item[this.itemValueKey]);
+			this.ngZone.run(() => {
+				if (this.type === "single" && !isUpdate(event)) {
+					this.closeMenu();
+					if (event.item && event.item.selected) {
+						if (this.itemValueKey) {
+							this.propagateChange(event.item[this.itemValueKey]);
+						} else {
+							this.propagateChange(event.item);
+						}
 					} else {
-						this.propagateChange(event.item);
+						this.propagateChange(null);
 					}
-				} else {
-					this.propagateChange(null);
 				}
-			}
 
-			if (this.type === "multi" && !isUpdate(event)) {
-				// if we have a `value` selector and selected items map them appropriately
-				if (this.itemValueKey && this.view.getSelected()) {
-					const values = this.view.getSelected().map(item => item[this.itemValueKey]);
-					this.propagateChange(values);
-				// otherwise just pass up the values from `getSelected`
-				} else {
-					this.propagateChange(this.view.getSelected());
+				if (this.type === "multi" && !isUpdate(event)) {
+					// if we have a `value` selector and selected items map them appropriately
+					if (this.itemValueKey && this.view.getSelected()) {
+						const values = this.view.getSelected().map(item => item[this.itemValueKey]);
+						this.propagateChange(values);
+					// otherwise just pass up the values from `getSelected`
+					} else {
+						this.propagateChange(this.view.getSelected());
+					}
 				}
-			}
-			// only emit selected for "organic" selections
-			if (!isUpdate(event)) {
-				this.checkForReorder();
-				this.selected.emit(event);
-			}
-			// manually tick the app so the view picks up any changes
-			this.appRef.tick();
+				// only emit selected for "organic" selections
+				if (!isUpdate(event)) {
+					this.checkForReorder();
+					this.selected.emit(event);
+				}
+			});
```

Comment:
Any specific reason why you switched from `ApplicationRef` to `NgZone`? Isn't `NgZone.Run` practically equivalent to ApplicationRef.tick() in this case?

View is a ContentChild & it's run _within_ Angular. We are subscribing to the select event & making any appropriate changes based on the stream of data.

Reply:
To be honest, I don't think we even need `ApplicationRef.tick()` here.... I think it might be there to fix bug caused by one of the Angular versions since we support from 7 to 15...

Reply:
>updated dropdown component to run code inside the Angular zone to force a change detection cycle to take place rather than calling tick method on the application reference as the tick call was throwing an error after the change to the dropdown list was introduced

That's interesting, what was the error thrown?

Reply:
<img width="778" alt="Screenshot 2023-01-31 at 1 36 58 PM" src="https://user-images.githubusercontent.com/18511181/215852183-4306e659-041c-46c8-8b5e-f8ca89703adf.png">

This was the error.

I could have used `setTimeout` instead of `ngZone.run` but it felt counter-productive to use `setTimeout` when I'm removing `setTimeout` in another area of the code which is the cause of the bug.

Reply:
From the slack thread I opened in the carbon help channel:

> Basically, I had to find an alternate solution to the one introduced in [809](https://github.com/carbon-design-system/carbon-components-angular/pull/809) because the appRef.tick call was throwing an error after I introduced the change in the dropdown list.
>
> I will say that the solution added by 809 and even the solution I have replaced it with are unconventional and there might be a deeper issue with the implementation of the dropdown component that is being masked.
>
> I say that because the combobox component has almost identical code with the subscribe block of the dropdown list select event emitter subscription yet it doesn't need an appRef.tick call to update the view.
>
> For clarity, I've linked the select subscribe block of the [dropdown component](https://github.com/carbon-design-system/carbon-components-angular/blob/842778a25d2db17fc19655f949dde4e4c32f08c6/src/dropdown/dropdown.component.ts#L389) and the [combo box component](https://github.com/carbon-design-system/carbon-components-angular/blob/842778a25d2db17fc19655f949dde4e4c32f08c6/src/combobox/combobox.component.ts#L482).